src/modules/Dashboard/features/Respondents/Popups/DataExportPopup/DataExportPopup_new.tsx

```diff
@@ -0,0 +1,254 @@
+import get from 'lodash.get';
+import { useFormContext } from 'react-hook-form';
+import { format } from 'date-fns';
+
+import { Modal } from 'shared/components/Modal';
+import { EnterAppletPassword } from 'shared/components/Password';
+import {
+  StyledBodyLarge,
+  StyledLinearProgress,
+  StyledModalWrapper,
+  theme,
+  variables,
+} from 'shared/styles';
+import { sendLogFile } from 'shared/utils';
+import { useSetupEnterAppletPassword } from 'shared/hooks';
+import { getExportDataApi } from 'api';
+import { getExportPageAmount } from 'modules/Dashboard/api/api.utils';
+import { DateFormats } from 'shared/consts';
+import { ExportDataFormValues } from 'shared/features/AppletSettings/ExportDataSetting/ExportDataSetting.types';
+
+import { DataExportPopupProps, ExecuteAllPagesOfExportData, Modals } from './DataExportPopup.types';
+import { AppletsSmallTable } from '../../AppletsSmallTable';
+import { useCheckIfHasEncryption } from '../Popups.hooks';
+import { ChosenAppletData } from '../../Respondents.types';
+import { getFormattedToDate } from './DataExportPopup.utils';
+import { useMultipleDecryptWorkers } from './DataExportPopup.hooks';
+import { BUFFER_PERCENTAGE } from './DataExportPopup.const';
+
+export const DataExportPopup = ({
+  filters = {},
+  popupVisible,
+  isAppletSetting,
+  setPopupVisible,
+  tableRows,
+  chosenAppletData,
+  setChosenAppletData,
+  'data-testid': dataTestid,
+}: DataExportPopupProps) => {
+  const { t } = useTranslation('app');
+  const { getValues } = useFormContext<ExportDataFormValues>() ?? {};
+  const { appletPasswordRef, submitForm } = useSetupEnterAppletPassword();
+  const [activeModal, setActiveModal] = useState(Modals.DataExport);
+  const abortControllerRef = useRef<AbortController | null>(null);
+
+  const appletId = get(chosenAppletData, isAppletSetting ? 'id' : 'appletId');
+  const subjectId = isAppletSetting ? undefined : (chosenAppletData as ChosenAppletData)?.subjectId;
+  const { encryption } = chosenAppletData ?? {};
+
+  const handleExportPopupClose = useCallback(() => {
+    setChosenAppletData?.(null);
+    setPopupVisible(false);
+  }, [setChosenAppletData, setPopupVisible]);
+
+  const {
+    dataIsExporting,
+    setDataIsExporting,
+    setExportDataQueue,
+    dataExportingApiRef,
+    limitRef,
+    finishedPagesRef,
+  } = useMultipleDecryptWorkers({
+    handleExportPopupClose,
+    appletId,
+    encryption,
+    filters,
+  });
+
+  const handleDataExportSubmit = async () => {
+    if (dataIsExporting || dataExportingApiRef.current || !appletId) {
+      return;
+    }
+
+    await executeAllPagesOfExportData({ appletId, targetSubjectIds: subjectId });
+  };
+
+  const hasEncryptionCheck = useCheckIfHasEncryption({
+    isAppletSetting,
+    appletData: chosenAppletData,
+    callback: handleDataExportSubmit,
+  });
+
+  const showEnterPwdScreen = !!chosenAppletData && !dataIsExporting && !hasEncryptionCheck;
+
+  const executeAllPagesOfExportData = useCallback(
```

Comment:
This also, gives me an impression of a separate class/function that could host this responsibility